checkov/common/bridgecrew/integration_features/features/licensing_integration.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, List, Optional
+
+from checkov.common.bridgecrew.code_categories import CodeCategoryType, CodeCategoryMapping
+from checkov.common.bridgecrew.integration_features.base_integration_feature import BaseIntegrationFeature
+from checkov.common.bridgecrew.licensing import BillingPlan, \
+    CategoryToSubscriptionMapping, CustomerSubscription
+from checkov.common.bridgecrew.platform_integration import bc_integration
+
+if TYPE_CHECKING:
+    from checkov.common.bridgecrew.platform_integration import BcPlatformIntegration
+    from checkov.common.output.report import Report
+    from checkov.common.typing import _BaseRunner
+
+
+LICENSE_KEY = 'platformLicense'
+MODULES_KEY = 'modules'
+BILLING_PLAN_KEY = 'billingPlan'
+
+
+class LicensingIntegration(BaseIntegrationFeature):
+    def __init__(self, bc_integration: BcPlatformIntegration) -> None:
+        super().__init__(bc_integration=bc_integration, order=6)
+        self.enabled_modules: List[CustomerSubscription] = []
+        self.billing_plan: Optional[BillingPlan] = None
+        self.open_source_only: bool = True
+
+    def is_valid(self) -> bool:
+        # We will always use this integration to determine what runs or not
+        return True
+
+    def pre_scan(self) -> None:
+        if not self.bc_integration.bc_api_key:
+            logging.debug('Running without API key, so only open source runners will be enabled')
+            self.open_source_only = True
+        elif not self.bc_integration.customer_run_config_response:
+            logging.debug('Customer run config response does not exist, but there is an API key, so there may be some integration issue. Proceeding with open source runners.')
+            self.open_source_only = True
+        else:
+            logging.debug('Found customer run config and using it for licensing')
+            license_details = self.bc_integration.customer_run_config_response.get(LICENSE_KEY)
+            logging.debug(f'User license details: {license_details}')
+
+            self.open_source_only = False
+            # the API will return True for all modules if they are on resource mode, so we don't actually need the billing plan explicitly here
+            self.enabled_modules = [CustomerSubscription(m) for m, e in license_details.get(MODULES_KEY).items() if e]
+            self.billing_plan = BillingPlan(license_details[BILLING_PLAN_KEY])
+
+    def is_runner_valid(self, runner: str) -> bool:
```

Comment:
@mikeurbanski1 Just making sure - we are leveraging here the runner's `check_type`, right?
Can we make that explicit, i.e.:
```suggestion
    def is_runner_valid(self, runner_check_type: str) -> bool:
```

(Need to change in other places too...)